reyna/src/com/b2msolutions/reyna/services/ForwardService.java

```diff
@@ -37,21 +57,25 @@ protected void onHandleIntent(Intent intent) {
                 return;
             }
 
-            Message message = this.repository.getNext();
+            if (!this.messageProvider.canSend()) {
+                Logger.v(TAG, "ForwardService: messageProvider cannot send");
+                return;
+            }
+
+            Message message = this.messageProvider.getNext();
             while(message != null) {
 
                 this.thread.sleep(SLEEP_MILLISECONDS);
 
                 Logger.v(TAG, "ForwardService: processing message " + message.getId());
```

Comment:
this is the last message id in the batch.
Will log the first and last message id in the batch provider.